bciers/apps/compliance/src/app/components/taskLists/1_manageObligationTaskList.ts

```diff
@@ -103,20 +110,22 @@ export const generateManageObligationTaskList: (
   ];
 
   let ggeaparInterestSection: TaskListElement[] = [];
+  const isObligationFullyPaid = Number(outstandingBalance) === 0;
 
-  if (hasLateSubmissionPenalty) {
+  if (isObligationFullyPaid && hasLateSubmissionPenalty) {
     ggeaparInterestSection = generateGgeaparInterestTaskList(
       complianceReportVersionId,
-      activePage === ActivePage.ReviewInterestSummary,
+      activePage,
     );
   }
 
   let automaticPenaltySection: TaskListElement[] = [];
 
   if (
-    Number(outstandingBalance) === 0 &&
-    (penaltyStatus === PenaltyStatus.NOT_PAID ||
-      penaltyStatus === PenaltyStatus.PAID)
+    isObligationFullyPaid &&
+    [PenaltyStatus.NOT_PAID, PenaltyStatus.PAID].includes(
+      penaltyStatus as PenaltyStatus,
+    )
```

Comment:
This logic is incomplete. It will show the Automatic Overdue penalty section even if one does not exist. This check needs to be similar to what is done above with the hasLateSubmissionPenalty boolean.
I think the schema in `bc_obps/compliance/schema/compliance_report_version.py` needs a resolver for has_overdue_penalty like what was done for has_late_submission_penalty & that value needs to be checked here.